# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/CallWorkerThread.h
# Commit: 1c0d9fdb8a06
# Full Hash: 1c0d9fdb8a06724c0166a9e08578e81952161b67
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2025-06-19 21:40:31
# Regressor Bug: 1968812
# File Overlap Count: 1
# Description:
#   Bug 1968812 - Remove the CallWorker thread and use TaskQueue::Observer instead. r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D252686
# ==============================================================================

diff -r a3b51595f6d1 -r 1c0d9fdb8a06 dom/media/webrtc/libwebrtcglue/CallWorkerThread.h
--- a/dom/media/webrtc/libwebrtcglue/CallWorkerThread.h	Thu Jun 19 08:18:29 2025 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,53 +0,0 @@
-/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim:set ts=2 sw=2 sts=2 et cindent: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef DOM_MEDIA_WEBRTC_LIBWEBRTCGLUE_CALLWORKERTHREAD_H_
-#define DOM_MEDIA_WEBRTC_LIBWEBRTCGLUE_CALLWORKERTHREAD_H_
-
-#include "mozilla/AbstractThread.h"
-#include "nsIDirectTaskDispatcher.h"
-#include "WebrtcTaskQueueWrapper.h"
-
-namespace mozilla {
-
-// Implements AbstractThread for running things on the webrtc TaskQueue.
-// Webrtc TaskQueues are not refcounted so cannot implement AbstractThread
-// directly.
-class CallWorkerThread final : public AbstractThread,
-                               public nsIDirectTaskDispatcher {
- public:
-  NS_DECL_THREADSAFE_ISUPPORTS
-  NS_DECL_NSIDIRECTTASKDISPATCHER
-
-  explicit CallWorkerThread(
-      UniquePtr<WebrtcTaskQueueWrapper<DeletionPolicy::NonBlocking>>
-          aWebrtcTaskQueue)
-      : AbstractThread(aWebrtcTaskQueue->mTaskQueue->SupportsTailDispatch()),
-        mWebrtcTaskQueue(std::move(aWebrtcTaskQueue)) {}
-
-  // AbstractThread overrides
-  nsresult Dispatch(already_AddRefed<nsIRunnable> aRunnable,
-                    DispatchReason aReason = NormalDispatch) override;
-  bool IsCurrentThreadIn() const override;
-  TaskDispatcher& TailDispatcher() override;
-  nsIEventTarget* AsEventTarget() override;
-  NS_IMETHOD
-  DelayedDispatch(already_AddRefed<nsIRunnable> aEvent,
-                  uint32_t aDelayMs) override;
-
-  NS_IMETHOD RegisterShutdownTask(nsITargetShutdownTask* aTask) override;
-  NS_IMETHOD UnregisterShutdownTask(nsITargetShutdownTask* aTask) override;
-
-  const UniquePtr<WebrtcTaskQueueWrapper<DeletionPolicy::NonBlocking>>
-      mWebrtcTaskQueue;
-
- protected:
-  ~CallWorkerThread() = default;
-};
-
-}  // namespace mozilla
-
-#endif